src/magic.cpp

```diff
@@ -1423,23 +1425,57 @@ int known_magic::get_spellname_max_width()
     return width;
 }
 
+static bool is_valid_invlet( const int invlet )
```

Comment:
Instead of these comparisons, you might consider borrowing from here:

https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/inventory.cpp#L40